src/call/github-activity-events-monitor.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+// Name         : githubActivityEventsMonitor
+// Description  : Indexes events pertaining to user activity to OpenSearch
+
+import { Probot } from 'probot';
+import { Resource } from '../service/resource/resource';
+import { validateResourceConfig } from '../utility/verification/verify-resource';
+import { OpensearchClient } from '../utility/opensearch/opensearch-client';
+
+export default async function githubActivityEventsMonitor(app: Probot, context: any, resource: Resource): Promise<void> {
+  if (!(await validateResourceConfig(app, context, resource))) return;
+
+  const repoName = context.payload.repository?.name;
+  const orgName = context.payload.organization?.login || context.payload.repository?.owner?.login;
+
+  const event = {
+    id: context.id,
+    organization: orgName,
+    repository: repoName,
+    type: context.name,
+    action: context.payload.action,
+    sender: context.payload.sender?.login,
+    created_at: new Date().toISOString(),
+  };
+
+  const client = await new OpensearchClient().getClient();
+
+  const [month, year] = [new Date().getMonth() + 1, new Date().getFullYear()].map((num) => String(num).padStart(2, '0'));
+
+  try {
+    await client.index({
+      index: `github-activity-events-${month}-${year}`,
```

Comment:
```suggestion
      index: `github-maintainers-activity-events-${month}-${year}`,
```

Reply:
Not sure if it should be named this, it is indexing all user activity, only when reindexing am I looking for maintainers. This way we can have data to look for candidate maintainers in the future. Although if there prove to be too many documents, I can index only the maintainer's events.

Reply:
In that case should we index all users activity, or just noop if user is not a maintainer?

Reply:
We should try all user activity as then we can tell if there's a user who's not currently not a maintainer who is contributing greatly, and potentially promote them to maintainer

Reply:
Though if there are too many documents and the cluster can't handle it, we can filter to index only the maintainers' events